src/main/java/graphs/Graph.java

```diff
@@ -1,60 +1,82 @@
 package graphs;
 
-import java.util.*;
 
-public class GraphClass {
-    private List<GraphEdge> edges;
-    private List<GraphNode> nodes;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.LinkedList;
+import java.util.Collections;
+
+
+public class Graph {
+    private List<Edge> edges;
+    private List<Node> nodes;
     private boolean isDirected;
     private boolean isWeighted;
 
-    public GraphClass() {
-        edges = new ArrayList<GraphEdge>();
-        nodes = new ArrayList<GraphNode>();
+    public Graph() {
+        edges = new ArrayList<Edge>();
+        nodes = new ArrayList<Node>();
         setDirected(false);
         setWeighted(false);
     }
 
-    public GraphClass(List<GraphNode> nodes, List<GraphEdge> edges) {
-        this.edges = edges;
+    public Graph(List<Node> nodes, List<Edge> edges) {
+        // if no flag for isDirected, default false
+        this(nodes, edges, false);
+    }
+
+    public Graph(List<Node> nodes, List<Edge> edges, boolean isDirected) {
         this.nodes = nodes;
-        setDirected(false);
+        setDirected(isDirected);
+        if(isDirected()) {
+            this.edges = edges;
+        } else {
+            this.edges = new ArrayList<Edge>();
+            for(Edge edge : edges) {
+                addEdge(edge);
+            }
+        }
         setWeighted(false);
     }
 
-    public GraphClass(GraphClass graph) {
+    public Graph(Graph graph) {
         edges = graph.getEdges();
```

Comment:
I dont really know why i implemented this, i guess just in case. here the new graph instance will have it's edges pointing to the same edges object of the first graph? so if i added an edge to the second graph, would the edges of the first graph update as well?